src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -194,6 +194,9 @@ bool convert_precision(ov::pass::PassBase& pass,
             bool is_output_precision_changed = false;
 
             for (auto& node : ops) {
+                // skip precision sensitive nodes
+                if (pass.transformation_callback(node))
+                    continue;
```

Comment:
why can't we add that check explicitly here instead callback?

Reply:
That's a good question! I also thought about that. But not all plugins support graphs with different precision. 
Some plugins can call `ConvertPrecission` expecting that it will convert the whole graph to `f16` so i think it's better to leave `ConvertPrecissions` behavior default as much as possible. So callback is less dangerous. And pass callbacks are introduced and are used in other transformation for this exact reason: to skip specific nodes.